packages/manager/apps/key-management-service/src/pages/listing/index.tsx

```diff
@@ -1,105 +1,137 @@
+  ODS_THEME_TYPOGRAPHY_SIZE,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+} from '@ovhcloud/ods-common-theming';
+import {
+  DataGridClipboardCell,
+  DataGridTextCell,
+  Datagrid,
+  Notifications,
+  useDatagridSearchParams,
+} from '@ovhcloud/manager-components';
+import { useOKMS } from '@/hooks/useOKMS';
+import { OKMS } from '@/interface';
 
 export default function Listing() {
-  const { t } = useTranslation('key-management-service/dashboard');
-  const [pageSize] = useState(10);
-  const {
-    data,
-    fetchNextPage,
-    isError,
-    hasNextPage,
-    isLoading,
-    error,
-    isFetching,
-  }: any = useInfiniteQuery({
-    initialPageParam: null,
-    queryKey: [`servicesListingIceberg`],
-    queryFn: ({ pageParam }) =>
-      getListingIcebergV2({ pageSize, cursor: pageParam }),
-    staleTime: Infinity,
-    getNextPageParam: (lastPage) => lastPage.cursorNext as any,
-  });
-
-  const tableContainerRef: React.MutableRefObject<HTMLDivElement> = React.useRef<
-    HTMLDivElement
-  >(null);
+  const { t } = useTranslation('key-management-service/listing');
+  const { t: tError } = useTranslation('error');
+  const navigate = useNavigate();
 
-  const flattenData = React.useMemo(
-    () => data?.pages?.flatMap((page: any) => page.data) ?? [],
-    [data],
-  );
-  const totalFetched = flattenData.length;
-
-  const fetchMoreOnBottomReached = React.useCallback(
-    (containerRefElement?: HTMLDivElement | null) => {
-      if (containerRefElement) {
-        const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
-        if (scrollHeight - scrollTop - clientHeight < 500 && !isFetching) {
-          fetchNextPage();
-        }
-      }
+  const columns = [
+    {
+      id: 'name',
+      cell: (props: OKMS) => {
+        return <DataGridTextCell>{props.iam.displayName}</DataGridTextCell>;
+      },
+      label: t('key_management_service_listing_name_cell'),
     },
-    [fetchNextPage, isFetching, totalFetched],
-  );
+    {
+      id: 'id',
+      cell: (props: OKMS) => {
+        return <DataGridClipboardCell text={props.id} />;
+      },
+      label: t('key_management_service_listing_id_cell'),
+    },
+    {
+      id: 'region',
+      cell: (props: OKMS) => {
+        return (
+          <DataGridTextCell>
+            {t(
+              `key_management_service_listing_region_${props.region.toLowerCase()}`,
+            )}
+          </DataGridTextCell>
+        );
+      },
+      label: t('key_management_service_listing_region_cell'),
+    },
+  ];
 
-  if (isError) {
-    return <ErrorBanner error={error} />;
-  }
+  const { sorting, setSorting } = useDatagridSearchParams();
 
-  if (isLoading) {
-    return (
-      <div>
-        <Loading />
-      </div>
-    );
-  }
+  const { error, data: okms, isLoading } = useOKMS({
+    sorting,
+  });
 
-  if (data?.length === 0) return <Navigate to="onboarding" />;
+  useEffect(() => {
+    if (okms.length === 0 && !isLoading) {
+      navigate('/onboarding');
```

Comment:
You should import your routes from a constant

Reply:
Fixed